### PR TITLE
Change the supervisor releases to require & use the semver field [part1]

### PR DESCRIPTION
**Pull request by thgreasi**
Connects-to: #811
Change-type: minor
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: src/features/supervisor-app/hooks/supervisor-app.ts [hunk 1]
```diff
@@ -48,15 +48,50 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
...
+			$select: 'id',
+			$filter: {
+				release_version: { $ne: null },
+			},
+		},
+	});
+
+	if (newSupervisorRelease == null) {
+		throw new BadRequestError(
+			`Could not find a supervisor release that has a release version with this ID ${newSupervisorReleaseId}`,
```
Comment (dfunckt) [inline src/features/supervisor-app/hooks/supervisor-app.ts:90]:
```suggestion
			`Supervisor release with ID ${newSupervisorReleaseId} does not exist or has no release version`,
```

**Review by dfunckt — COMMENTED** [review]
Had a quick look, looks good to me.

File: src/features/supervisor-app/hooks/supervisor-app.ts (hunk 1 above)
Comment (thgreasi) [reply src/features/supervisor-app/hooks/supervisor-app.ts:90]: :+1:

**Comment by thgreasi** [comment]
@balena-ci I self-certify!

**Pull request merged**